File: models/Gpt-3.5-turbo/py/code/additional_problems/Permutations_II.py

MOD = 10**9 + 7

def countBeautifulPermutations(n):
    dp = [[0] * (n+1) for _ in range(n+1)]

    for j in range(1, n+1):
        dp[1][j] = 1

    for i in range(2, n+1):
        for j in range(1, n+1):
            for k in range(1, n+1):
                if abs(j-k) > 1:
                    dp[i][j] = (dp[i][j] + dp[i-1][k]) % MOD

    ans = sum(dp[n]) % MOD
    return ans

# Test the code
n = int(input())
print(countBeautifulPermutations(n))

File: models/Gpt-3.5-turbo/cpp/code/problems/additional_problems/Permutations_II.cpp

#include <iostream>
#include <vector>

const int MOD = 1e9 + 7;

int countBeautifulPermutations(int n) {
    std::vector<int> dp(n + 1);
    dp[0] = 1;

    for (int i = 1; i <= n; ++i) {
        if (i >= 2) {
            dp[i] = (dp[i] + dp[i - 2]) % MOD;
        }
        dp[i] = (dp[i] + dp[i - 1]) % MOD;
    }

    return dp[n];
}

int main() {
    int n;
    std::cin >> n;

    std::cout << countBeautifulPermutations(n) << std::endl;

    return 0;
}
